snorkel/labeling/apply/lf_applier.py

```diff
@@ -0,0 +1,44 @@
+from itertools import chain
+from typing import Any, Iterable, List, Tuple
+
+import scipy.sparse as sparse
+from tqdm import tqdm
+
+from snorkel.labeling.lf import LabelingFunction
+from snorkel.types import Example
+
+RowData = List[Tuple[int, int, int]]
+
+
+class BaseLFApplier:
+    def __init__(self, lfs: List[LabelingFunction]) -> None:
+        if not all(isinstance(f, LabelingFunction) for f in lfs):
```

Comment:
N00b question here (I've been out of the loop for a bit...): Are there any generic commands / tools for enforcing type checking automatically, given the type annotations?  Seems redundant to have explicit checks like this?

Reply:
A very cursory check: I see a few libs for letting you add a decorator to enforce types at runtime (https://github.com/RussBaz/enforce, https://github.com/Stewori/pytypes).  Don't know what the "standard" is though.  Perhaps we could just spin our own, just so that this is a decorator e.g. `@enforcetype` instead of this manual check?

Reply:
This check is completely unnecessary. I'll remove it. The onus is on the user to use the API correctly, and they'll find out pretty quickly if they don't. Even more quickly if they use mypy.

Reply:
Well I think it's nice to have it.  And more generally, it also seems nice to be able to add a decorator that says "if types were defined, enforce them at runtime", so we should have this?

Reply:
Let's talk offline